src/test/kotlin/uk/gov/justice/digital/hmpps/nomisvisitsmappingservice/resource/IncentiveMappingResourceIntTest.kt

```diff
@@ -120,6 +120,43 @@ class IncentiveMappingResourceIntTest : IntegrationTestBase() {
       ).isEqualTo("Validation failure: Incentive mapping id = 4444 already exists")
     }
 
+    @Test
+    internal fun `create mapping succeeds when the same mapping already exists for the same IEP`() {
+      webTestClient.post().uri("/mapping/incentives")
+        .headers(setAuthorisation(roles = listOf("ROLE_NOMIS_INCENTIVES")))
+        .contentType(MediaType.APPLICATION_JSON)
+        .body(
+          BodyInserters.fromValue(
+            """{
+            "nomisBookingId"     : $bookingId,
+            "nomisIncentiveSequence"     : $sequence,
+            "incentiveId"      : $incentiveId,
+            "label"       : "2022-01-01",
+            "mappingType" : "INCENTIVE_CREATED"
+          }"""
+          )
+        )
+        .exchange()
+        .expectStatus().isCreated
+
+      webTestClient.post().uri("/mapping/incentives")
+        .headers(setAuthorisation(roles = listOf("ROLE_NOMIS_INCENTIVES")))
+        .contentType(MediaType.APPLICATION_JSON)
+        .body(
+          BodyInserters.fromValue(
+            """{
+            "nomisBookingId"     : $bookingId,
+            "nomisIncentiveSequence"     : $sequence,
+            "incentiveId"      : $incentiveId,
+            "label"       : "2022-01-01",
+            "mappingType" : "INCENTIVE_CREATED"
+          }"""
+          )
+        )
+        .exchange()
+        .expectStatus().isCreated
```

Comment:
I suppose 200 would have been better, but the mess in the code is just not worth it